tests/unit/style/parsing/effects.rs

```diff
@@ -106,3 +107,8 @@ fn test_parse_factor() {
     assert!(parse(filter::parse, "sepia(-1)").is_err());
     assert!(parse(filter::parse, "saturate(-1)").is_err());
 }
+
+fn blur_radius_should_not_accept_negavite_values() {
+    use style::properties::longhands::box_shadow;
+    assert!(parse(box_shadow::parse, "1px 1px -1px").is_err());
```

Comment:
Please add a test to ensure that it does parse when the value is valid, e.g. `1px 1px 0px` and `1px 1px 1px`, so that it can be more apparently that the value is rejected because of the negative value, not anything else.